pkg/meta/base.go

```diff
@@ -818,7 +819,39 @@ func (m *baseMeta) HandleQuota(ctx Context, cmd uint8, dpath string, quotas map[
 			}
 			quotas[p] = quota
 		}
-	default: // FIXME: QuotaCheck
+	case QuotaCheck:
+		q, err := m.en.doGetQuota(ctx, inode)
+		if err != nil {
+			return err
+		}
+		if q == nil {
+			return fmt.Errorf("no quota for inode %d path %s", inode, dpath)
+		}
+		var sum Summary
+		if st := m.GetSummary(ctx, inode, &sum, true, strict); st != 0 {
+			return st
+		}
+		usedInodes := int64(sum.Dirs+sum.Files) - 1
+		usedSpace := int64(sum.Size) - align4K(0) // quota ignore root dir
+		if q.UsedInodes == usedInodes && q.UsedSpace == usedSpace {
+			logger.Infof("quota of %s is consistent", dpath)
+			quotas[dpath] = q
+			return nil
+		}
+		logger.Errorf(
+			"%s: quota(%s, %s) != summary(%s, %s)", dpath,
+			humanize.Comma(q.UsedInodes), humanize.IBytes(uint64(q.UsedSpace)),
+			humanize.Comma(usedInodes), humanize.IBytes(uint64(usedSpace)),
+		)
+		if repair {
+			q.UsedInodes = usedInodes
+			q.UsedSpace = usedSpace
+			quotas[dpath] = q
+			logger.Info("reparing...")
+			return m.en.doSetQuota(ctx, inode, q, false)
```

Comment:
`doSetQuota` will not update the used stats

Reply:
It seems the used stats will be updated if the `create` flag is true.

Reply:
done